File: KKProject/Controller/KKProfile/KKProfileViewController.m
//
//  KKProfileViewController.m
//  KKProject
//
//  Created by Macbook Pro 15.4  on 2019/5/1.
//  Copyright © 2019 Macbook Pro 15.4 . All rights reserved.
//

#import "KKProfileViewController.h"
#import "KKSettingsViewController.h"

@interface KKProfileViewController ()


@end

@implementation KKProfileViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}

- (void)kk_settingsEvent:(UIBarButtonItem *)item{
    KKSettingsViewController *vc = KKSettingsViewController.new;
    vc.kk_navTitle = @"设置";
    [self.navigationController pushViewController:vc animated:true];
}

- (void)kk_layoutNavigation{
    [super kk_layoutNavigation];
    self.kk_navShadowImage        = UIImage.new;
    self.kk_navShadowColor        = [UIColor clearColor];
    self.kk_navBackgroundColor    = UIColor.clearColor;
    self.kk_navRightBarButtonItem = [UIBarButtonItem kk_itemWithTitle:nil
                                                           titleColor:nil
                                                            imageName:[UIImage imageNamed:@"profile_settings.png"]
                                                        highImageName:[UIImage imageNamed:@"profile_settings.png"]
                                                               target:self
                                                               action:@selector(kk_settingsEvent:)];
}



/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: KKProject_Swift/Profile/KKProfileViewController.swift
//
//  KKProfileViewController.swift
//  KKProject_Swift
//
//  Created by youbin on 2020/1/20.
//  Copyright © 2020 Macbook Pro 15.4 . All rights reserved.
//

import UIKit

class KKProfileViewController: KKViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        

        let rxSwiftButton : UIButton  = UIButton.init(type: .custom)
        rxSwiftButton.backgroundColor = .red
        rxSwiftButton.frame = CGRect.init(x: 0, y: 0, width: 44, height: 44)
        
        self.navigationItem.rightBarButtonItem = UIBarButtonItem.init(customView: rxSwiftButton)
    
        
      
        // Do any additional setup after loading the view.
    }
    
    
    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
        self.navigationController?.pushViewController(KKRxSwiftViewController(), animated: true)
    }

    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        // Get the new view controller using segue.destination.
        // Pass the selected object to the new view controller.
    }
    */

}
